### PR TITLE
Fix imdbid regex

### PR DESCRIPTION
**Pull request by randrey**
**Changes**
Updated regex to allow more than 7 digits in imdb id. 

**Issues**
#2902

### CODE REVIEW

File: Emby.Server.Implementations/Library/PathExtensions.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public static string GetAttributeValue(this string str, string attrib)
             // for imdbid we also accept pattern matching
             if (string.Equals(attrib, "imdbid", StringComparison.OrdinalIgnoreCase))
             {
-                var m = Regex.Match(str, "tt\\d{7}", RegexOptions.IgnoreCase);
+                var m = Regex.Match(str, "tt\\d{7,}", RegexOptions.IgnoreCase);
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Library/PathExtensions.cs:42]: AFAIK the id can only be either 7 or 8 digits
```suggestion
                var m = Regex.Match(str, "tt\\d{7,8}", RegexOptions.IgnoreCase);
```

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 2]
```diff
@@ -208,8 +208,8 @@ protected virtual void Fetch(MetadataResult<T> item, string metadataFile, XmlRea
 
         protected void ParseProviderLinks(T item, string xml)
         {
-            // Look for a match for the Regex pattern "tt" followed by 7 digits
-            var m = Regex.Match(xml, @"tt([0-9]{7})", RegexOptions.IgnoreCase);
+            // Look for a match for the Regex pattern "tt" followed by 7 or more digits
+            var m = Regex.Match(xml, @"tt(\d{7,})", RegexOptions.IgnoreCase);
```
Comment (mark-monteiro) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:212]: Same

File: Emby.Server.Implementations/Library/PathExtensions.cs (hunk 1 above)
Comment (randrey) [reply Emby.Server.Implementations/Library/PathExtensions.cs]: Updated.

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 3]
```diff
@@ -208,8 +208,8 @@ protected virtual void Fetch(MetadataResult<T> item, string metadataFile, XmlRea
 
         protected void ParseProviderLinks(T item, string xml)
         {
-            // Look for a match for the Regex pattern "tt" followed by 7 digits
-            var m = Regex.Match(xml, @"tt([0-9]{7})", RegexOptions.IgnoreCase);
+            // Look for a match for the Regex pattern "tt" followed by 7 or 8 digits
+            var m = Regex.Match(xml, @"tt(\d{7,8})", RegexOptions.IgnoreCase);
```
Comment (mark-monteiro) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:212]: I found this interesting point about using `\d` vs `[0-9]`, so I think we actually want `[0-9]` here: https://stackoverflow.com/a/16621778/1988326

Confirmed locally as well:
image

Comment (Bond-009) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:212]: We should probably change all regexes that use `\d` and document his somewhere

Comment (joshuaboniface) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:212]: Is this within the scope of the PR though? I agree though @randrey we should probably leave it as `[0-9]` for this reason, unless this change in behaviour is part of the fix.

Comment (Bond-009) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:212]: Changing all other regexes should ofc be done in a separate pr

Comment (mark-monteiro) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:212]: Created #2923 to track updating all other regexes.

Comment (randrey) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs]: That's an interesting observation. 
I've updated the code in those two files to use [0-9].

**Pull request merged**